improver_tests/utilities/test_compare.py

```diff
@@ -300,3 +374,78 @@ def message_collector(message):
                          100.0, 100.0, message_collector)
     assert len(messages_reported) == 1
     assert "shape" in messages_reported[0]
+
+
+def test_compare_extra_dimension(dummy_nc):
+    """Check an additional (but unused) dimension is reported"""
```

Comment:
This test also covers mismatched dimension sizes. It could either be a separate test or should be mentioned in the doc-string.